using_images/s2i_images/customizing_s2i_images.adoc

```diff
@@ -69,10 +70,9 @@ The example shows a custom *_assemble_* script that prints the message,
 executes standard *_assemble_* script from the image and prints another message
 depending on the exit code of the *_assemble_* script.
 
-Wrapping the *_run_* script is a bit tricky because we have to use `exec` for
-invoking it (see chapter
-xref:../../creating_images/guidelines.adoc#general-docker-guidelines[Use `exec` in
-Wrapper Scripts]). As a consequence, it is impossible to have a post-run step
+Wrapping the *_run_* script is a bit tricky because you have to
+xref:../../creating_images/guidelines.adoc#general-docker-guidelines[use `exec`
+for invoking it]. As a consequence, it is impossible to have a post-run step
 because the code after `exec /usr/libexec/s2i/run` never gets executed.
```

Comment:
one thing being tricky (running commands before the existing run script) leads to another thing being impossible (running things after the existing run script).  but i agree it can use a rephrasing.

better would be:
When wrappering the run script you must use exec for invoking it to ensure signals are handled properly.  Unfortunately the use of exec also precludes the ability to run additional commands after invoking the default image run script.